nemo/collections/tts/data/vocabs.py

```diff
@@ -0,0 +1,240 @@
+            self.oov, self.blank = self.blank, self.oov
+
+        self.labels = labels
+        self.sep = sep
+
+        self._util_ids = {self.pad, self.blank, self.oov}
+        self._label2id = {l: i for i, l in enumerate(labels)}
+        self._id2label = labels
+
+    def __call__(self, text: str) -> List[int]:
+        return self.encode(text)
+
+    @abc.abstractmethod
+    def encode(self, text: str) -> List[int]:
+        """Turns str text into int tokens."""
+        pass
+
+    def decode(self, tokens: List[int]) -> str:
+        """Turns ints tokens into str text."""
+        return self.sep.join(self._id2label[t] for t in tokens if t not in self._util_ids)
+
+
+class EnglishChars(Base):
+    """English char-based vocabulary."""
+
+    # fmt: off
+    PUNCT_LIST = (  # Derived from LJSpeech and "/" additionally
+        ',', '.', '!', '?', '-',
+        ':', ';', '/', '"', '(',
+        ')', '[', ']', '{', '}',
+    )
+    # fmt: on
+
+    def __init__(
+        self,
+        punct=True,
+        spaces=False,
+        apostrophe=True,
+        add_blank_at="last_but_one",
+        pad_with_space=False,
+        non_default_punct_list=None,
+        text_preprocessing_func=english_text_preprocessing,
+        word_tokenize_func=english_word_tokenize,
+    ):
+        labels = []
+        self.space, labels = len(labels), labels + [' ']  # Space
+        labels.extend(string.ascii_lowercase)
+        if apostrophe:
+            labels.append("'")  # Apostrophe for saving "don't" and "Joe's"
+
+        if punct:
+            if non_default_punct_list is not None:
+                self.PUNCT_LIST = non_default_punct_list
+            labels.extend(self.PUNCT_LIST)
+
+        super().__init__(labels, add_blank_at=add_blank_at)
+
+        self.punct = punct
+        self.spaces = spaces
+
+        self.pad_with_space = pad_with_space
+
+        self.text_preprocessing_func = text_preprocessing_func
+        self.word_tokenize_func = word_tokenize_func
+
+    def encode(self, text):
+        """See base class."""
+        cs, space, labels = [], self.labels[self.space], set(self.labels)
+
+        for c in "".join(self.word_tokenize_func(self.text_preprocessing_func((text)))):  # noqa
+            # Add space if last one isn't one
+            if c == space and len(cs) > 0 and cs[-1] != space:
+                cs.append(c)
+
+            # Add next char
+            if (c.isalnum() or c == "'") and c in labels:
+                cs.append(c)
+
+            # Add punct and remove space if needed
+            if (c in self.PUNCT_LIST) and self.punct:
+                if not self.spaces and len(cs) > 0 and cs[-1] == space:
+                    cs.pop()
+                cs.append(c)
+
+        # Remove trailing spaces
+        while cs[-1] == space:
+            cs.pop()
+
+        if self.pad_with_space:
+            cs = [space] + cs + [space]
+
+        return [self._label2id[p] for p in cs]
+
+
+class EnglishPhonemes(Base):
```

Comment:
Can you add a test for this class? One with some inputs (challenging preferred) and expected output to track any changes that might happen to this class.